### PR TITLE
Add commands to move to top/bottom frame

### LINKED ISSUE #298: Improve frame command to move between frames
Would it be possible to add a `frame` command that allows to jump between frames as in https://stackoverflow.com/a/37402920/2149400 ?

My main use case is during post mortem debugging where pdb opens at the oldest/original frame (the innermost exception) and I have to press `up` several times to reach the conflicting line in my code. More details in this so question: https://stackoverflow.com/a/37402920/2149400

### PR DESCRIPTION
**Pull request by petobens**
Closes #298

### CODE REVIEW

File: pdb.py [hunk 1]
```diff
@@ -1457,6 +1457,29 @@ def do_down(self, arg='1'):
     do_down.__doc__ = pdb.Pdb.do_down.__doc__
     do_d = do_down
 
+    def do_top(self, arg):
+        if self.curindex == 0:
+            print('*** Oldest frame', file=self.stdout)
```
Comment (blueyed) [inline pdb.py]: Could use `self.error()`, and return (for less indent below).
Same below.

**Review by blueyed — COMMENTED** [review]
Looking at https://github.com/fschulze/pytest-pdb/blob/master/pytest_pdb.py#L97-L113 shows there's a `_select_frame` method that could be used apparently?

**Comment by blueyed** [comment]
Re coverage: due to `set_trace` it is usually expected to not be covered afterwards, so don't worry.

File: pdb.py (hunk 1 above)
Comment (petobens) [reply pdb.py]: Changed in https://github.com/pdbpp/pdbpp/pull/307/commits/7f3073c9f520e672a8bc331ddf52a5a30d084ebd

**Comment by petobens** [comment]
> Looking at https://github.com/fschulze/pytest-pdb/blob/master/pytest_pdb.py#L97-L113 shows there's a _select_frame method that could be used apparently?

I tried this (with top) and it doesn't seem to have the same behaviour (in the sense that the current implementation is identical to running `up` multiple times whereas if I use `_select_frame(0)` it takes me elsewhere)

**Comment by blueyed** [comment]
Will look into this, can you allow me to push here (see checkbox on the right)?

**Comment by blueyed** [comment]
> I tried this (with top) and it doesn't seem to have the same behaviour (in the sense that the current implementation is identical to running `up` multiple times whereas if I use `_select_frame(0)` it takes me elsewhere)

Appears to work for me - can you clarify "elsewhere"?
Check `bt` for when that happens.

**Comment by blueyed** [comment]
```diff
 pdb.py              | 17 ++++-------------
 testing/test_pdb.py | 29 ++++++++++++++++++++++++++++-
 2 files changed, 32 insertions(+), 14 deletions(-)

diff --git c/pdb.py w/pdb.py
index 74ece24..8e8ac13 100644
--- c/pdb.py
+++ w/pdb.py
@@ -1458,28 +1458,19 @@ def do_down(self, arg='1'):
     do_d = do_down
 
     def do_top(self, arg):
+        """Go to top (oldest) frame."""
         if self.curindex == 0:
             self.error('Oldest frame')
             return
-        else:
-            self.curindex = 0
-            self.curframe = self.stack[self.curindex][0]
-            self.curframe_locals = self.curframe.f_locals
-            self.print_current_stack_entry()
-            self.lineno = None
+        self._select_frame(0)
     do_top = do_top
 
-
     def do_bottom(self, arg):
+        """Go to bottom (newest) frame."""
         if self.curindex + 1 == len(self.stack):
             self.error('Newest frame')
             return
-        else:
-            self.curindex = len(self.stack) - 1
-            self.curframe = self.stack[self.curindex][0]
-            self.curframe_locals = self.curframe.f_locals
-            self.print_current_stack_entry()
-            self.lineno = None
+        self._select_frame(len(self.stack) - 1)
     do_bottom = do_bottom
 
     @staticmethod
diff --git c/testing/test_pdb.py w/testing/test_pdb.py
index 39efafc..3df4c0c 100644
--- c/testing/test_pdb.py
+++ w/testing/test_pdb.py
@@ -1019,13 +1019,40 @@ def c():
    5 frames hidden .*
 # top
 [NUM] > .*()
--> sys.exit(main())
+-> .*main
 # bottom
 [NUM] > .*c()
 -> return
 # c
 """)
 
+
+def test_top_bottom_post_mortem():
+    def fn():
+        def throws():
+            0 / 0
+
+        try:
+            throws()
+        except:
+            pdb.post_mortem(Pdb=PdbTest)
+    check(fn, r"""
+[1] > .*throws()
+-> 0 / 0
+# top
+[0] > .*fn()
+-> throws()
+# top
+\*\*\* Oldest frame
+# bottom
+[1] > .*throws()
+-> 0 / 0
+# bottom
+\*\*\* Newest frame
+# c
+""")
+
+
 def test_parseline():
     def fn():
         c = 42
```

**Comment by petobens** [comment]
> Will look into this, can you allow me to push here (see checkbox on the right)?

Feel free to add any changes/ commit you deem fit (I checked again and the behaviour seems indeed the same)

**Comment by blueyed** [comment]
Done.
What happened to the idea of using `f 0` / `f -1` instead?  (the latter needs fixing)
We could still have `top` / `bottom` though for clarity, of course.

**Comment by petobens** [comment]
Thanks for picking this up! Looks great

**Comment by blueyed** [comment]
Great, merged in 069da80.
Thanks again!

**Pull request closed without merging**